# Informe EDA

In [69]:
%run lib.ipynb

# Lectura del dataset de películas
df_movies = pd.read_parquet("datasets/movies_dataset.parquet")
df_movies.info()
df_movies.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45376 entries, 0 to 45375
Data columns (total 21 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   belongs_to_collection  4488 non-null   object 
 1   budget                 45376 non-null  object 
 2   genres                 45376 non-null  object 
 3   id                     45376 non-null  object 
 4   original_language      45365 non-null  object 
 5   overview               44435 non-null  object 
 6   popularity             45376 non-null  float64
 7   production_companies   45376 non-null  object 
 8   production_countries   45376 non-null  object 
 9   release_date           45376 non-null  object 
 10  revenue                45376 non-null  float64
 11  runtime                45130 non-null  float64
 12  spoken_languages       45376 non-null  object 
 13  status                 45296 non-null  object 
 14  tagline                20398 non-null  object 
 15  ti

belongs_to_collection    budget  \
0  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1                                               None  65000000   
2  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3                                               None  16000000   
4  {'id': 96871, 'name': 'Father of the Bride Col...         0   

                                              genres     id original_language  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...    862                en   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   8844                en   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...  15602                en   
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...  31357                en   
4                     [{'id': 35, 'name': 'Comedy'}]  11862                en   

                                            overview  popularity  \
0  Led by Woody, Andy's toys live happily in his ...   21.946943   
1  When siblings Judy and Peter discover an encha...   17.015539   
2  A family wedding reignites the ancient feud be...   11.712900   
3  Cheated on, mistreated and stepped on, the wom...    3.859495   
4  Just when George Banks has recovered from his ...    8.387519   

                                production_companies  \
0     [{'name': 'Pixar Animation Studios', 'id': 3}]   
1  [{'name': 'TriStar Pictures', 'id': 559}, {'na...   
2  [{'name': 'Warner Bros.', 'id': 6194}, {'name'...   
3  [{'name': 'Twentieth Century Fox Film Corporat...   
4  [{'name': 'Sandollar Productions', 'id': 5842}...   

                                production_countries release_date  ...  \
0  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-10-30  ...   
1  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-15  ...   
2  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22  ...   
3  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22  ...   
4  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-02-10  ...   

   runtime                                   spoken_languages    status  \
0     81.0           [{'iso_639_1': 'en', 'name': 'English'}]  Released   
1    104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...  Released   
2    101.0           [{'iso_639_1': 'en', 'name': 'English'}]  Released   
3    127.0           [{'iso_639_1': 'en', 'name': 'English'}]  Released   
4    106.0           [{'iso_639_1': 'en', 'name': 'English'}]  Released   

                                             tagline  \
0                                               None   
1          Roll the dice and unleash the excitement!   
2  Still Yelling. Still Fighting. Still Ready for...   
3  Friends are the people who let you be yourself...   
4  Just When His World Is Back To Normal... He's ...   

                         title vote_average  vote_count  release_year  \
0                    Toy Story          7.7      5415.0          1995   
1                      Jumanji          6.9      2413.0          1995   
2             Grumpier Old Men          6.5        92.0          1995   
3            Waiting to Exhale          6.1        34.0          1995   
4  Father of the Bride Part II          5.7       173.0          1995   

      return  populartiy  
0  12.451801   21.946943  
1   4.043035   17.015539  
2   0.000000   11.712900  
3   5.090760    3.859495  
4   0.000000    8.387519  

[5 rows x 21 columns]

## Chequeo de nulos

In [70]:
#suma de valores nulos por columna
print(columnas_de_nulos(df_movies))


belongs_to_collection    40888
original_language           11
overview                   941
runtime                    246
status                      80
tagline                  24978
dtype: int64


### De un total de 45376, la columna "belongs_to_collection" tiene un 90.1% de valores nulos, como además la columna no será utilizada en el API (ni en el recomendador), será eliminada

In [71]:
df_movies = df_movies.drop(['belongs_to_collection'], axis=1)
print(columnas_de_nulos(df_movies))

original_language       11
overview               941
runtime                246
status                  80
tagline              24978
dtype: int64


### Para la columna "original_language", tenemos 11 filas con valor nulo. 8 de ellos tienen sólo un valor en la columna "spoken_languages" así que lo imputaremos en "original_languaje", para las 4 restantes imputaremos el primer valor que se encuentre en #spoken_languajes". Esta columna es de interes ya que posteriormente la utilizaré para reducir el dataset y poder factibilizar la ejecución del algoritmo de recomendación, ya que con todo el dataset genera problemas de memoria.

In [72]:
df_movies.loc[[21579, 22805, 32103], ['original_language']] = 'en'
df_movies.loc[[19559, 44494], ['original_language']] = ''
df_movies.loc[[37354], ['original_language']] = 'cs'
df_movies.loc[[40982], ['original_language']] = 'ur'
df_movies.loc[[41802, 44573], ['original_language']] = 'xx'
df_movies.loc[[43978], ['original_language']] = 'fr'
df_movies.loc[[44328], ['original_language']] = 'sv'

print(columnas_de_nulos(df_movies))

overview      941
runtime       246
status         80
tagline     24978
dtype: int64


### Utilizaré la columna "overview" como la base del algoritmo de recomendación, por lo que necesito que este poblada para todas las películas. Utilizaré en primera instancia la columna "tagline" y posteriormente la columna "title".

In [73]:

#Reemplazo los valores nulos de la columna overview por su correspondiente en tagline
df_movies["overview"] = df_movies["overview"].fillna(df_movies["tagline"])

#Reemplazo los valores nulos de la columna overview por su correspondiente en title
df_movies["overview"] = df_movies["overview"].fillna(df_movies["title"])

print(columnas_de_nulos(df_movies))

runtime      246
status        80
tagline    24978
dtype: int64


## Valores duplicados

## Valores faltantes

## Tipos de datos

## Análisis estadístico

## Análisis de correlaciones